src/MacroTools/Mechanics/DemonGates/DemonGateType.cs

```diff
@@ -20,11 +21,13 @@ public sealed class DemonGateType : PassiveAbility
     /// <param name="demonUnitTypeId">The unit to spawn.</param>
     /// <param name="spawnInterval">How often to spawn the unit.</param>
     /// <param name="spawnCount">How many of the unit to spawn each time.</param>
-    public DemonGateType(int gateUnitTypeId, int demonUnitTypeId, float spawnInterval, int spawnCount) : base(gateUnitTypeId)
+    /// <param name="spawnLimit">How many of the unit to spawn each time.</param>
+    public DemonGateType(int gateUnitTypeId, int demonUnitTypeId, float spawnInterval, int spawnCount, int spawnLimit = 12) : base(gateUnitTypeId)
```

Comment:
Just take out the default IMO. The caller should know what limit they're looking for - there's not many to configure.
```suggestion
    public DemonGateType(int gateUnitTypeId, int demonUnitTypeId, float spawnInterval, int spawnCount, int spawnLimit) : base(gateUnitTypeId)
```

Reply:
Amended